File: ssafybottle_server/src/main/java/com/ssafy/cafe/model/dao/NotificationDao.java
package com.ssafy.cafe.model.dao;

import java.util.List;
import com.ssafy.cafe.model.dto.Notification;

public interface NotificationDao {
	
	List<Notification> showNoti(String userId);
	
	int insert(Notification noti);
	
	int deleteAll(String userId);
	
	int delete(int id);
}

File: ssafybottle_user/app/src/main/java/com/ssafy/smartstore/data/local/dao/NotificationDao.kt
package com.ssafy.smartstore.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.ssafy.smartstore.data.entitiy.Notification

@Dao
interface NotificationDao {
    @Query("SELECT * FROM t_notification WHERE user_id = (:userId)")
    suspend fun getNotifications(userId: String) : List<Notification>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertNotification(notification: Notification)

    @Delete
    suspend fun deleteNotification(notification: Notification) : Int

    @Query("DELETE FROM t_notification WHERE user_id = (:userId)")
    suspend fun deleteAllNotifications(userId: String) : Int
}